### ISSUE TITLE
Mac build: cmake error: imported target includes non-existent path

### ISSUE THREAD

**Issue by ogallagher** [fontforge/fontforge#4926]
## Environment

FontForge v[0bf72v2](https://github.com/fontforge/fontforge/commit/0bf72b2f8a9ba5c784e23586c5b8dfc0658d9d89) (**master** as of 2022-03-05).

Mac OS Catalina v10.15.7

Homebrew v3.4.0

## CMake error during build

![cmake error imported target includes non-existent path](https://user-images.githubusercontent.com/17031438/156902019-1cf3253f-8c01-43be-9d3e-9c6a6eb86c16.png)

I encountered the above errors when following the [install.md](https://github.com/fontforge/fontforge/blob/master/INSTALL.md) instructions.

## Hacked workaround

The font forge build files were asking for two library locations that I didn’t have:

- `/usr/local/Cellar/glib/2.68.3/include/glib-2.0`
- `/usr/local/Cellar/gtk+3/3.24.32/include/gtk-3.0`

However, I did have newer versions of both of these libraries:

- `/usr/local/Cellar/glib/2.70.4/include/glib-2.0`
- `/usr/local/Cellar/gtk+3/3.24.33/include/gtk-3.0`

So I faked it by making sibling symlinks with the requested versions:

```bash
# symlink glib
cd /usr/local/Cellar/glib/
ln -s 2.70.4 2.68.3

# confirm glib
ls -l
# total 8
# lrwxr-xr-x  <stuff> 2.68.3 -> 2.70.4
# drwxr-xr-x  <stuff> 2.70.4

# symlink gtk
cd /usr/local/Cellar/gtk+3/
ln -s 3.24.33 3.24.32

# confirm gtk
ls -l
# total 8
# lrwxr-xr-x  <details> 3.24.32 -> 3.24.33
# drwxr-xr-x  <details> 3.24.33
```

**Comment by ogallagher** [comment]
I can also confirm that, using the homebrew library version symlinks hack, I am able build and then run `fontforge` successfully.

**Comment by ferdnyc** [comment]
@ogallagher
Were you using an old, existing build directory to (re-)configure the FontForge build?

The most likely explanation is that you were running `cmake` in a `build` directory which had previously been configured, and Homebrew upgraded the dependencies between the configuration runs. FontForge's build generation will discover the locations of various system libraries using `pkg-config`, which on macOS with Homebrew most likely points it at the actual `Cellar` directories those files reside in. Then, when the build environment is generated again, the cached paths are used and CMake doesn't know that the files have been moved.

(Either that, or this is a bug in the packaging for `glib-2.0` and `gtk+3` that should be reported to Homebrew. But I don't think that's the case.)

You should be able to get an up-to-date build config by renaming or deleting your existing `build/` subdirectory and generating into an empty one, i.e.:
```sh
cd /path/to/fontforge
rm -r build
mkdir build
cd build
cmake -GNinja ..
```

If you don't want to blow away your entire build directory, you can try deleting _just_ the file `build/CMakeCache.txt` and then re-running the `cmake -GNinja ..` line. That'll cause CMake to re-discover all of its dependencies, and will _probably_ also work. (But completely removing the old `build` directory is the safer, more effective approach.)

**Comment by ferdnyc** [comment]
The step-by-step version of that second suggestion would be

```sh
cd /path/to/fontforge
cd build
rm CMakeCache.txt
cmake -GNinja ..
```

**Comment by ferdnyc** [comment]
The other possibility I considered was that you might have a `/path/to/fontforge/CMakeCache.txt` file or a `/path/to/fontforge/CMakeFiles/` directory in the source repo (_not_ inside a `build/` subdirectory), which can happen if you ever generate a build system in the source dir accidentally.  Those build files can cause incredibly difficult-to-debug problems later on, even when running CMake in a clean build directory.

But that _shouldn't_ be a problem here, because FontForge's CMake code would error out if an in-source build is attempted. That happens long before anything important gets configured. So, while it can leave a `/path/to/fontforge/CMakeCache.txt` file behind, there's hardly anything in it and it isn't likely to cause troubles.

Yeah, my money's on the dirty/out-of-date `build/` directory.

**Comment by ogallagher** [comment]
> Were you using an old, existing build directory to (re-)configure the FontForge build?
> 
> The most likely explanation is that you were running cmake in a build directory which had previously been configured, and Homebrew upgraded the dependencies between the configuration runs.

@ferdnyc Yep, that was the problem. I tried running `cmake` earlier, then saw I had missing dependencies, installed them (probably upgrading some existing ones, namely `glib` and `gtk`), and ran `cmake` again without cleaning the already generated files in `build/`. Thanks for explaining! 

I just confirmed that I could clean the `build/` dir, remove my renumbered library aliases, and then rebuild successfully.

Is this mistake worth mentioning in `INSTALL.md`? I could write it as I’ve already forked this repo, but someone else could too, of course.
